ext/simplexml/simplexml.c

```diff
@@ -277,7 +277,7 @@ static zval *sxe_prop_dim_read(zval *object, zval *member, zend_bool elements, z
 		if (!member && node && node->parent &&
 		    node->parent->type == XML_DOCUMENT_NODE) {
 			/* This happens when the user did: $sxe[]->foo = $value */
-			php_error_docref(NULL, E_ERROR, "Cannot create unnamed attribute");
+			zend_throw_error(zend_ce_error, "Cannot create unnamed attribute");
```

Comment:
For reviewers: As this is behind a `!member` guard the string copy in L259 couldn't have happened, so we don't have to free it.